chore: update job queue table

hamidzr opened the pull request:
## Description

- improve how job preemptability is rendered for fairshare scheduler.
- improve unreleased manage job modal visual and update logic

<!---
Lead with the intended commit body in this description field. For breaking changes,
please include "BREAKING CHANGE:" at the beginning of your commit body.
At a minimum, this section should include a bracketed reference to the Jira ticket,
e.g. "[DET-1234]". When squash-and-merging, copy this directly into the description field.
-->


## Test Plan

<!---
Describe the situations in which you've tested your change, and/or a screenshot as appropriate.
Reviewers may ask questions about this test plan to ensure adequate manual coverage of changes.
-->


## Commentary (optional)

![2022-01-04_13-44-32](https://user-images.githubusercontent.com/12127420/148127890-d73ee3b3-3c41-4c4a-b48b-30502c5e9210.png)
![2022-01-04_13-43-19](https://user-images.githubusercontent.com/12127420/148127896-eafde945-890f-482e-978b-a801de69d78d.png)


<!---
Use this section of your description to add context to the PR. Could be for particularly
tricky bits of code that could use extra scrutiny, historical context useful for reviewers, etc.
You may intentionally leave this section blank and remove the title.
--->


## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.


<!---
## Title

Example title: "docs: tweak recommended "pip install" usage".

Specifically, this title should contain a type and a description
of the change being made:

User-facing change types:
- docs: docs-only change
- feat: new user-facing feature
- fix: bug fix
- perf: performance improvement

Internal change types:
- build: build system change (anything in a `Makefile`, mostly)
- chore: any internal change not covered by another type
- ci: anything that touches `.circleci`
- refactor: internal refactor
- style: style change
- test: new tests

See https://www.conventionalcommits.org/en/v1.0.0/ for background.

The first line should also:
- be at most 89 characters long
- contain a description that is at most 72 characters long
- not end with sentence-ending punctuation
- start (after the type) with a lowercase imperative ("add", "fix")
-->


[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hkang1 commented on webui/react/src/components/Table.tsx:
```diff
@@ -59,8 +59,8 @@ export const defaultPaginationConfig = {
 
-export const archivedRenderer = (archived: boolean): React.ReactNode => {
-  return archived ? <Icon name="checkmark" /> : null;
+export const checkmarkRenderer = (yesNo: boolean): React.ReactNode => {
```
smart

hkang1 commented on webui/react/src/pages/JobQueue/JobQueue.table.tsx:
```diff
@@ -77,7 +77,7 @@ export const columns: ColumnType<Job>[] = [
       const TypeCell = (
         <Tooltip placement="topLeft" title={title}>
-          <div className={''}>
+          <div>
```
good find

hkang1 commented on webui/react/src/utils/job.ts:
```diff
@@ -51,3 +52,7 @@ export const jobStateToLabel: {[key in JobState]: string} = {
   [JobState.QUEUED]: 'Queued',
 };
+
+export const orderedSchedulers = new Set(
+  [ Api.V1SchedulerType.PRIORITY, Api.V1SchedulerType.KUBERNETES ],
+);
```
**question:** would you mind adding `utils/job.test.ts` with some tests for these utility functions? I forgot to mention it last time 🙏

hamidzr replied:
thanks for the reminder.

hamidzr commented:
there was an[ unrelated cli failure in CI](https://app.circleci.com/pipelines/github/determined-ai/determined/17911/workflows/9622d241-89d7-43e4-b28c-f94c6ed1cc37/jobs/549316) for the last run I"m going to update the branch.

hamidzr commented on webui/react/src/utils/job.ts:
```diff
@@ -31,23 +33,30 @@ export const jobTypeToCommandType = (
+);
+
+export const moveJobToPositionUpdate = (jobId: string, position: number): Api.V1QueueControl => {
+  if (position < 1 || position % 1 !== 0) {
```
probably not that useful 🤔

hkang1 commented:
Jotting the notes we discussed today:

When `IS_DEV` is Not set, encountered the following:
* User shows `Unavailable (U)` for forked experiments (CLI works ok with the correct user, but forking a new CLI experiment still causes user `Unavailable`)
* Job name is not coming up properly after a rename via the experiment detail header UI.
* -1 priority shows up still
* Preemptible column shows on priority scheduler (expected behavior of being hidden)

hamidzr commented:
thanks for the review. 

> User shows Unavailable (U) for forked experiments (CLI works ok with the correct user, but forking a new CLI experiment still causes user Unavailable)

backend problem exposed by WebUI moving to the new create experiment API. getting fixed here https://github.com/determined-ai/determined/issues/3389.
This should not happen when creating experiments from the cli even without the work linked above.

> Job name is not coming up properly after a rename via the experiment detail header UI.

Another backend issue tracked here https://github.com/determined-ai/determined/issues/3390

> -1 priority shows up still
> Preemptible column shows on priority scheduler (expected behavior of being hidden)

these last two were from a missing boolean negation I had changed last minute. fixed here https://github.com/determined-ai/determined/pull/3374/commits/b8b86ddd0c64a15262c255ea0dd24a8a2a34fe10

The backend issues are separate from the goal of the pr and WebUI and I think we can move forward w/o waiting for them if all the changes in the PR are strictly improvements.

hkang1 commented on webui/react/src/utils/job.ts:
```diff
@@ -31,23 +33,30 @@ export const jobTypeToCommandType = (
+
+export const moveJobToPositionUpdate = (jobId: string, position: number): Api.V1QueueControl => {
+  if (position < 1 || position % 1 !== 0) {
+    const err: DaError = {
+      message: `Invalid queue position: ${position}`,
+      type: ErrorType.Input,
+    };
+    throw err;
```
**suggestion:** don't worry about this as part of this PR. `DaError` is probably best as a class that wraps around `Error` so that it can inherit everything from `Error`, and it can be declared and used in the same manner as `Error`.

e.g. `throw new DaError({ message: ..., type: ... })`

hamidzr replied:
yeah, that'd work nicely. 👍🏼

hkang1 reviewed (approved):
LGTM! Thanks for addressing the other issues in separate PRs.

hamidzr commented:
> LGTM! Thanks for addressing the other issues in separate PRs.

thanks for helping catch them. one of them is still in progress but it wouldn't affect this pr.
